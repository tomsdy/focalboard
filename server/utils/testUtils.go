package utils

import "github.com/stretchr/testify/mock"

var Anything = mock.MatchedBy(func(interface{}) bool { return true })
