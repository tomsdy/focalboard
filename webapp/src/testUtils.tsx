// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {IntlProvider} from 'react-intl'
import React from 'react'

export const wrapIntl = (children: any) => <IntlProvider locale='en'>{children}</IntlProvider>

export function mockDOM(): void {
    window.focus = jest.fn()
    document.createRange = () => {
        const range = new Range()

        range.getBoundingClientRect = jest.fn()

        range.getClientRects = () => {
            return {
                item: () => null,
                length: 0,
                [Symbol.iterator]: jest.fn(),
            }
        }

        return range
    }
}
