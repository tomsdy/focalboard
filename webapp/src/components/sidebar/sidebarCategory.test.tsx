// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react'

import {createMemoryHistory} from 'history'
import {Router} from 'react-router-dom'

import {render} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import {Provider as ReduxProvider} from 'react-redux'

import configureStore from 'redux-mock-store'

import {TestBlockFactory} from '../../test/testBlockFactory'

import {wrapIntl} from '../../testUtils'

import SidebarCategory from './sidebarCategory'

describe('components/sidebarCategory', () => {
    const board = TestBlockFactory.createBoard()

    const view = TestBlockFactory.createBoardView(board)
    view.fields.sortOptions = []
    const history = createMemoryHistory()

    const board1 = TestBlockFactory.createBoard()
    const board2 = TestBlockFactory.createBoard()
    const boards = [board1, board2]
    const categoryBlocks1 = TestBlockFactory.createCategoryBlocks()
    categoryBlocks1.name = 'Category 1'
    categoryBlocks1.blockIDs = [board1.id, board2.id]

    const categoryBlocks2 = TestBlockFactory.createCategoryBlocks()
    categoryBlocks2.name = 'Category 2'

    const categoryBlocks3 = TestBlockFactory.createCategoryBlocks()
    categoryBlocks3.name = 'Category 3'

    const allCategoryBlocks = [
        categoryBlocks1,
        categoryBlocks2,
        categoryBlocks3,
    ]

    const state = {
        users: {
            me: {
                id: 'user_id_1',
            },
        },
        boards: {
            current: board.id,
            boards: {
                [board.id]: board,
            },
        },
        views: {
            current: view.id,
            views: {
                [view.id]: view,
            },
        },
        teams: {
            current: {
                id: 'team-id',
            },
        },
    }

    test('sidebar call hideSidebar', () => {
        const mockStore = configureStore([])
        const store = mockStore(state)

        const component = wrapIntl(
            <ReduxProvider store={store}>
                <Router history={history}>
                    <SidebarCategory
                        hideSidebar={() => {}}
                        categoryBlocks={categoryBlocks1}
                        boards={boards}
                        allCategories={allCategoryBlocks}
                    />
                </Router>
            </ReduxProvider>,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()

        // testing collapsed state of category
        const subItems = container.querySelectorAll('.category > .IconButton')
        expect(subItems).toBeDefined()
        userEvent.click(subItems[0] as Element)
        expect(container).toMatchSnapshot()
    })
})
