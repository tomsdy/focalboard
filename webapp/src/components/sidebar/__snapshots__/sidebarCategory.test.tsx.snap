// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sidebarCategory sidebar call hideSidebar 1`] = `
<div>
  <div
    class="SidebarCategory"
  >
    <div
      class="octo-sidebar-item category ' expanded "
    >
      <button
        class="IconButton"
        type="button"
      >
        <i
          class="CompassIcon icon-chevron-down ChevronDownIcon"
        />
      </button>
      <div
        class="octo-sidebar-title category-title"
        title="Category 1"
      >
        Category 1
      </div>
      <div
        aria-label="menuwrapper"
        class="MenuWrapper"
        role="button"
      >
        <button
          class="IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-dots-horizontal OptionsIcon"
          />
        </button>
      </div>
    </div>
    <div
      class="SidebarBoardItem subitem "
    >
      <div
        class="octo-sidebar-icon"
      >
        i
      </div>
      <div
        class="octo-sidebar-title"
        title="board title"
      >
        board title
      </div>
      <div
        aria-label="menuwrapper"
        class="MenuWrapper x"
        role="button"
      >
        <button
          class="IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-dots-horizontal OptionsIcon"
          />
        </button>
      </div>
    </div>
    <div
      class="SidebarBoardItem subitem "
    >
      <div
        class="octo-sidebar-icon"
      >
        i
      </div>
      <div
        class="octo-sidebar-title"
        title="board title"
      >
        board title
      </div>
      <div
        aria-label="menuwrapper"
        class="MenuWrapper x"
        role="button"
      >
        <button
          class="IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-dots-horizontal OptionsIcon"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/sidebarCategory sidebar call hideSidebar 2`] = `
<div>
  <div
    class="SidebarCategory"
  >
    <div
      class="octo-sidebar-item category ' collapsed "
    >
      <button
        class="IconButton"
        type="button"
      >
        <i
          class="CompassIcon icon-chevron-right ChevronRightIcon"
        />
      </button>
      <div
        class="octo-sidebar-title category-title"
        title="Category 1"
      >
        Category 1
      </div>
      <div
        aria-label="menuwrapper"
        class="MenuWrapper"
        role="button"
      >
        <button
          class="IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-dots-horizontal OptionsIcon"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
